pandas/core/apply.py

```diff
@@ -178,6 +177,228 @@ def apply(
         """
 
 
+class NumbaExecutionEngine(BaseExecutionEngine):
+    """
+    Numba-based execution engine for pandas apply and map operations.
+    """
+
+    @staticmethod
+    def map(
+        data: np.ndarray | Series | DataFrame,
+        func,
+        args: tuple,
+        kwargs: dict,
+        decorator: Callable | None,
+        skip_na: bool,
+    ):
+        """
+        Elementwise map for the Numba engine. Currently not supported.
+        """
+        raise NotImplementedError(
+            "The Numba engine is not implemented for the map method yet."
+        )
+
+    @staticmethod
+    def apply(
+        data: np.ndarray | Series | DataFrame,
+        func,
+        args: tuple,
+        kwargs: dict,
+        decorator: Callable,
+        axis: int | str,
+    ):
+        """
+        Apply `func` along the given axis using Numba.
+        """
+
+        NumbaExecutionEngine.check_numba_support(func)
+
+        # normalize axis values
+        if axis in (0, "index"):
+            axis = 0
+        else:
+            axis = 1
+
+        # check for data typing
+        if not isinstance(data, np.ndarray):
+            if data.empty:
+                return data.copy()  # mimic apply_empty_result()
+            NumbaExecutionEngine.validate_values_for_numba_raw_false(
+                data, decorator if isinstance(decorator, dict) else {}
```

Comment:
It'd be good to make the second argument a keyword argument, it's not clear from this call what the method is expecting.

And I'm not sure when `decorator` can be a dictionary? Afaik `decorator` is the engine, I don't understand the logic here. Is it possible that this always pass an empty dictionary? Feels like if you want to check if numba.jit has been initialized with `parallel=True` or others you should check an attribute of it.